packages/uui-text-copy/lib/uui-text-copy.element.ts

```diff
@@ -0,0 +1,175 @@
+   * @type {"default" | "primary" | "secondary" | "outline" | "placeholder"}
+   * @attr
+   * @default "default"
+   */
+  @property()
+  look: UUIInterfaceLook = 'default';
+
+  /**
+   * Changes the look of the button to one of the predefined, symbolic looks. For example - set this to positive if you want nice, green "confirm" button.
+   * @type {"default" | "positive" | "warning" | "danger"}
+   * @attr
+   * @default "default"
+   */
+  @property({ reflect: true })
+  color: UUIInterfaceColor = 'default';
+
+  /**
+   * Makes the left and right padding of the button narrower.
+   * @type {boolean}
+   * @attr
+   * @default false
+   */
+  @property({ type: Boolean })
+  compact = false;
+
+  /**
+   * The delay in milliseconds before the button returns to its default state after a successful copy
+   * @type {number}
+   * @attr
+   * @default 250
+   */
+  @property({ type: Number, attribute: 'animation-state-delay' })
+  animationStateDelay = 250;
+
+  constructor() {
+    super();
+    demandCustomElement(this, 'uui-button');
+    demandCustomElement(this, 'uui-icon');
+  }
+
+  // Used to store the value that will be copied to the clipboard
+  #valueToCopy = '';
+
+  readonly #onClick = async (e: Event) => {
+    const button = e.target as UUIButtonElement;
+    button.state = 'waiting';
+
+    // By default use the value property
+    this.#valueToCopy = this.value;
+
+    // If copy-from is set use that instead
+    if (this.copyFrom) {
+      // Try & find an element with the ID
+      const el = document.getElementById(this.copyFrom);
+      if (el) {
+        console.log('Element found to copy from', el);
+        this.#valueToCopy = el.textContent ?? el.innerText ?? '';
+
+        // Override the value to copy ,if the element has a value property
+        // Such as uui-input or uui-textarea or native inout elements
+        if ('value' in el) {
+          console.log('This element has a value property', el);
+          this.#valueToCopy = (el as any).value;
+        }
+      } else {
+        console.error(`Element ID ${this.copyFrom} not found to copy from`);
+        button.state = 'failed';
+        return;
+      }
+    }
+
+    const beforeCopyEv = new UUITextCopyEvent(UUITextCopyEvent.COPYING, {
+      detail: { text: this.#valueToCopy },
+    });
+    this.dispatchEvent(beforeCopyEv);
+
+    if (beforeCopyEv.detail.text != null) {
+      this.#valueToCopy = beforeCopyEv.detail.text;
+    }
+
+    await navigator.clipboard
+      .writeText(this.#valueToCopy)
+      .then(() => {
+        this.dispatchEvent(
+          new UUITextCopyEvent(UUITextCopyEvent.COPIED, {
+            detail: { text: this.#valueToCopy },
+          }),
+        );
+        setTimeout(() => {
+          button.state = 'success';
+        }, this.animationStateDelay);
+      })
+      .catch(err => {
+        button.state = 'failed';
+        console.error('Error copying to clipboard', err);
+      });
```

Comment:
Consider making a try/catch statement around this rather than `then` and `catch`.